File: UI/src/app/models/insurance-contract.ts
import {CompensationModel} from "./compensation";

export interface InsuranceContractModel {
  insuranceContractId: string,
  consumerId: string,
  insuranceAgent: string,
  startDate: Date,
  endDate: Date,
  compensations: CompensationModel[],
  insurancePremium: number,
  insuranceAmount: number,
  state: InsuranceContractState,
  agentCompanyName: string,
}

export enum InsuranceContractState {
  Created,
  Processing,
  Approved,
  Rejected
}

File: Core/Entities/InsuranceContract.cs
namespace Core.Models;

public class InsuranceContract
{
    public Guid Id { get; set; }
    public Consumer Consumer { get; set; }
    public InsuranceAgent InsuranceAgent { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public IEnumerable<Compensation>? Compensations { get; set; }
    public decimal InsurancePremium { get; set; }
    public decimal InsuranceAmount { get; set; }
}